src/resolver/article.resolver.ts

```diff
@@ -0,0 +1,82 @@
+import { Resolver, Query, Arg, Args, Mutation } from "type-graphql";
+import { Article, ArticleStatus } from "../../common/models";
+import { TableProvider } from "../mongodb/mongodb-manager";
+import { ArticleTable } from "./article-table";
+
+@Resolver(Article)
+export class ArticleResolver extends TableProvider(ArticleTable) {
+    @Query(() => [Article], { nullable: false })
+    public async allArticles(): Promise<Article[]> {
+        const tbl = await this.table();
+        return await tbl.getAllArticles();
+    }
+
+    @Query(() => Article, { nullable: true })
+    public async article(
+        @Arg('id') id: string,
+    ): Promise<Article | null> {
+        const tbl = await this.table();
+        return await tbl.getArticle(id);
+    }
+
+    @Query(() => [Article], { nullable: true })
+    public async articles(
+        @Arg('ids', () => [String]) ids: string[],
+    ): Promise<Article[] | null> {
```

Comment:
Can we have a function to get all articles without providing ids?

Reply:
yes! the getAllArticles can do that

Reply:
oh great! did not see that